src/index.js

```diff
@@ -161,29 +165,96 @@ module.exports = function ({ types: t }) {
     if (node.__coverage__instrumented) return
     node.__coverage__instrumented = true
 
-    const id = nextStatementId(context, node.loc)
+    const id = nextStatementId(context, path)
     instrument(path, increase(context, 's', id))
   }
 
   //
+  // check a path to see if it has a ignore comment
+  // set the zone
+  //
+  function checkIgnoredPath (path, data) {
+    path.node.leadingComments && path.node.leadingComments.some((comment) => {
```

Comment:
Using `&&` just for condition and `Array#some` just for breakable loop, I think it makes the code less straightforward. Can you do something like this instead?

```js
    if (!path.node.leadingComments) return
    for (const comment of path.node.leadingComments) {
      // ...
      if (match) {
        // ...
        return
      }
    }
```

Using for-of, we also need to enable babel-plugin-runtime to ensure that it still works in Node v0.12.